### ISSUE TITLE
Files created when rendering are deleted

### ISSUE THREAD

**Issue by aimundo** [quarto-dev/quarto-cli#105]
Hi, I have started to use Quarto with RStudio and really like its functionalities. I am currently creating a Quarto file to show the analysis of some data and I have noticed that if I use a Windows 10 Desktop, when rendering the "intermediate files" (PNG files in this case from the R figures), those files can be accessed as they are in a "-files" folder. This is useful because I can use the PNG files for a presentation or something else, and  the quality is good.

But when I render using  a laptop with Windows 10, the folder that hosts those files is deleted upon completion of the rendering.  And I am using the exact same Quarto file in  each case, which is puzzling.

Could you provide some support on why this is happening?

**Comment by jjallaire** [comment]
If you have `self-contained` set for the laptop one then it will delete the file, but I'm at a loss as to why it would be deleted otherwise. What happens if you add `keep-md` to the metadata for the file?

**Comment by jjallaire** [comment]
It could be that on the laptop you are actually using R Markdown (not
Quarto). The two differ in default behavior (Quarto keeps _files and R
Markdown makes self-contained HTML so deletes it). Does the rendered HTML
look identical?

**Comment by aimundo** [comment]
I'm a little confused by the question about RMarkdown. My understanding is that I am using Quarto in both cases as my file is a `*.qmd` file, which won't be recognized by the RStudio IDE that is not the Preview Version. The rendered HTML looks identical.

**Comment by aimundo** [comment]
Sure thing. Here is the file rendered from the desktop.

**Comment by jjallaire** [comment]
I don't think that's working for the attachment (you likely. need to use the GitHub web interface to do the attachment). Also, I really need to see the file from the laptop not the desktop (as that's the one that is having problems). Generally I just need more information and data than just "it doesn't work on the laptop" as we don't have a known failure to work correctly on laptops.

**Comment by aimundo** [comment]
Sorry, it took me a while to get back to this. I had to make another file because the information on the original file is sensitive, but this one below has the same basic structure. I rendered it in my laptop and I see the same issue of not keeping the intermediate files. I hope this helps.

[GAM_models_GitHub_issue.zip](https://github.com/quarto-dev/quarto-cli/files/7112121/GAM_models_GitHub_issue.zip)

**Comment by jmbuhr** [comment]
This contains just one html-file (`GAM_models_GitHub_issue.html`) did you want to include the source document (like the qmd, Rmd or ipynb-file) instead? I also see that you are loading a lot of packages, you might have better luck with a more minimal reproducible example.

**Comment by jjallaire** [comment]
We do indeed need to see the source document. The file you attached was
definitely rendered using `self-contained` mode. This could have
happened based on YAML in the file, YAML in a Quarto project file
(including one up the directory hierarchy from your source file) or could
have come from a user-level pandoc defaults file somewhere in your home
directory. I think you get this to _stop_ happening by adding
self-contained: false to the YAML for your source file, but it would be
nice to know why it's happening in the first place.

**Comment by aimundo** [comment]
[GAM_models_GitHub_issue.zip](https://github.com/quarto-dev/quarto-cli/files/7117806/GAM_models_GitHub_issue.zip)

I see. Here is the original file, I removed most of the additional packages. This is the same file I rendered in my laptop the other day. The weird thing is that I just rendered it my desktop and the folder with the intermediate files is being kept.

**Comment by jjallaire** [comment]
Yeah when I render this the intermediate files are indeed kept. Quarto
outputs a bunch of YAML in the terminal when you do a render (you'd see it
perhaps in the Render or Jobs tab in RStudio). What does that YAML look
like? (it's debug info that indicates what values were passed to pandoc).
If there is no evidence of self-contained there then there is some other
source of this behavior in your configuration (could be pandoc settings)
causing this.

**Comment by aimundo** [comment]
Below is what I found in "Jobs" (in my PC). I would have to render the file in the laptop later to see how that YAML looks like.
--
pandoc 
  to: html
  from: markdown
  output-file: GAM_models_GitHub_issue.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  css:
    - style.css
  toc: true
  number-sections: true
  filters:
    - crossref
  
metadata
  document-css: false
  link-citations: true
  title: '**DRS GAMs FOR COMPLETE DATA SET V1**'
  author: Ariel Mundo
  editor_options:
    chunk_output_type: console
  
Output created: GAM_models_GitHub_issue.html

**Comment by jjallaire** [comment]
That's the right content but we need that for the laptop.

You also may want to execute this command:

```
$ quarto pandoc --version
pandoc 2.14.2
Compiled with pandoc-types 1.22, texmath 0.12.3.1, skylighting 0.11,
citeproc 0.5, ipynb 0.1.0.1
User data directory: /Users/jjallaire/.local/share/pandoc
Copyright (C) 2006-2021 John MacFarlane. Web:  https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

And see if there are any files in the "User data directory" that is printed.

Finally, it would be good to know the exact version of RStudio you are using on the laptop.

(base) jjallaire@jjallaires-mbp quarto-web %

**Comment by jjallaire** [comment]
Looking again at the HTML generated on the laptop, I can tell you with 100% certainly that that HTML file was created with the rmarkdown package rather than quarto. It could be that you have an older version of RStudio that is not triggering the usage of quarto, but the content of the file was definitely not created using quarto. Most likely reasons for this are either (a) An older version of RStudio or (b) RStudio can't find the installation of Quarto and is falling back to using the rmarkdown package.

**Comment by aimundo** [comment]
I was finally able to try this on the laptop. Something weird is going on, because in the laptop there is nothing on the "Jobs" tab when the document is rendered. I am using the pre-release build of RStudio:
```
"Ghost Orchid" Preview (8aea4f32, 2021-08-27) for Windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.8 Chrome/69.0.3497.128 Safari/537.36
```

However, on the "Render" tab this is what appears after rendering:

```
"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS GAM_models_GitHub_issue.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output GAM_models_GitHub_issue.html --lua-filter "C:\Users\aimundo\R\win-library\4.0\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "C:\Users\aimundo\R\win-library\4.0\rmarkdown\rmarkdown\lua\latex-div.lua" --self-contained --variable bs3=TRUE --standalone --section-divs --template "C:\Users\aimundo\R\win-library\4.0\rmarkdown\rmd\h\default.html" --no-highlight --variable highlightjs=1 --variable theme=bootstrap --include-in-header "C:\Users\aimundo\AppData\Local\Temp\Rtmp4MSFf5\rmarkdown-str13d43c794cac.html" --mathjax --variable "mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" 
```

I do see the `--self-contained` in the output above. But then this is even more strange. Why, even I do have the latest RStudio release and I have the package `quarto` both in the PC and laptop there is a configuration issue? I have also attached the laptop html output below.
[GAM_models_GitHub_issue (2).zip](https://github.com/quarto-dev/quarto-cli/files/7140743/GAM_models_GitHub_issue.2.zip)

**Comment by jjallaire** [comment]
For whatever reason rmarkdown is being used to render that file on the laptop. Could be because Quarto is not in fact installed or (or findable from RStudio?). What is the output of `Sys.which("quarto")` on the laptop?

**Comment by jmbuhr** [comment]
I may have some insights on this. When `Sys.which` is called, it is not called from a login-shell. This means that `which` will not know about any `PATH` variables defined e.g. in `bashrc` like `~/bin/`, where quarto puts itself if it is not allowed to write to `/usr/bin/`. I don't know about the windows equivalent, but maybe RStudio can also not find user-installed, only system-wide quarto?

**Comment by jjallaire** [comment]
What is the exact PATH where `quarto` is installed on the laptop?

What is the result of both `Sys.which("quarto")` and `Sys.getenv("PATH")` from the R session?
